### ISSUE TITLE
[v2] Definition for rule 'jsx-a11y/href-no-hash' was not found

### ISSUE THREAD

**Issue by kripod** [gatsbyjs/gatsby#5411]
## Description

When eslint-plugin-jsx-a11y@6 is installed in a project using gatsby@2, I get the following console warning for every single page:

```
(node:2244) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
 WARNING  Compiled with 1 warnings                                                  19:07:36


C:\Development\Projects\gatsby-v2-test\src\pages\index.jsx
  1:1  warning  Definition for rule 'jsx-a11y/href-no-hash' was not found  jsx-a11y/href-no-hash

✖ 1 problem (0 errors, 1 warning)


You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file.
‼ ｢wdm｣:
i ｢wdm｣: Compiled with warnings.
```

### Steps to reproduce

Set up a new gatsby@2 project, run `yarn add -D eslint-plugin-jsx-a11y` and then run `gatsby develop`.

### Environment
* Gatsby version (`npm list gatsby`): 2.0.0-alpha.32
* gatsby-cli version (`gatsby --version`): 2.0.0-alpha.32
* Node.js version: 8.11.1
* Operating System: Windows 10 x64

### File contents (if changed)

`package.json`:

```json
{
  "private": true,
  "scripts": {
    "build": "gatsby build",
    "develop": "gatsby develop"
  },
  "dependencies": {
    "gatsby": "^2.0.0-alpha.32",
    "react": "^16.3.2",
    "react-dom": "^16.3.2"
  },
  "devDependencies": {
    "eslint-plugin-jsx-a11y": "^6.0.3"
  }
}
```

**Comment by kripod** [comment]
A simple temporary workaround would be updating eslint-config-react-app to [3.0.0-next.66cc7a90](https://unpkg.com/eslint-config-react-app@3.0.0-next.66cc7a90/index.js) (from [2.1.0](https://unpkg.com/eslint-config-react-app@2.1.0/index.js)).

**Comment by kripod** [comment]
Fixed by #5419
